Tweaks on Design 2.0

MaximeKjaer opened the pull request:
Hi Nargren!
You've done some great work, the UI looks amazing :D 

I've added one feature, station_cnt (see http://piratebox.aod-rpg.de/dokuwiki/doku.php?id=modification_list#show_connected_clients), to all the files, and I thought I'd share it, so that other users can benefit from it!
I have also updated the readme.md with noob-friendly instructions, to reach all PirateBox owners and I have added a section or two, plus minor design tweaks.

I hope you like it!

Maxime

MaximeKjaer committed:
Typo corrected, features added

Removed code which was unnecessary (line 10 and 11: The chat is already reloading in chat.html)
Added station_cnt.txt support (refreshes every minute.
Added mobile unit detection and redirection
Corrected a typo (line 70: punds) and set the price to 35£
Added a section showing what the PirateBox is composed of (line 76)


station_cnt is a feature which shows how many users are connected.
See: http://piratebox.aod-rpg.de/dokuwiki/doku.php?id=modification_list#show_connected_clients

MaximeKjaer committed:
Update Design2.0/index.html

MaximeKjaer committed:
Added ::selection, #station_cnt and #station_cnt a

Added selection style
Added rules for div#station_cnt and the anchor tag itself.

MaximeKjaer committed:
Added ::selection, #station_cnt and #station_cnt a

MaximeKjaer committed:
Update Design2.0/mindex.html

Added station_cnt support
Added support for browser version (line 28)
"Nothing illegal" is no longer in caps, and is like the browser version

MaximeKjaer committed:
Typo and station_cnt support

Corrected a typo (it's now contents instead of contens)
Added styling for station_cnt

MaximeKjaer committed:
Updated the readme with more precise instructions

Added exact commands needed for installing and deleting the UI.
Added Credits (Thanks, Nargren! My PBox looks way cooler now :D )

Nargren commented:
Hi Maxime!

Sorry for getting back to you only now, but I am getting a little short on time.
First of all, thank you very much for the improvements and modifications! 

Adding station_cnt to display connected users is a really nice improvement and I also like the automatic redirect to the mobile website. (I was trying to make the latter to work myself as well, but didn't have too much success.) Thanks for making these work!

Hopefully I will have some time tomorrow and over the weekend, I will quickly upgrade my box with your improvements, run through the code and pull your changes to my repository.

You can also contact me at: nargren@gmail.com

Thanks for the improvements again and I will get back to you soon.

Nargren

Nargren commented on Design2.0/index.html:
```diff
@@ -6,11 +6,24 @@
 	<script src="jquery.min.js"/>	
 	<script type="text/javascript">
+	
+	if (/Android|webOS|iPhone|iPod|BlackBerry/i.test(navigator.userAgent) && localStorage["mobile"] != "false")
```
I am not familiar with mobile redirects, but this didn't work for my BlackBerry device with the MR3020. I will look into it again and see if I can find the issue.

MaximeKjaer replied:
This could actually be 2 things:
1. Your BlackBerry's user agent string doesn't match any of these (which would be odd considering that BlackBerry is on that list).
2. Your device does not support localStorage (it's HTML5), resulting in an error. I have never seen anything like that, but I'll try to see how a device that doesn't support HTML5 behaves. We still have two options: cookies instead of localStorage, and then to put the whole thing in an if-statement: if (localStorage) { /_Check userAgent_/ }. I'll do some testing to find out.

Nargren commented:
Hello Maxime,

I downloaded the improved files from your repository and uploaded them to my PB on the MR3020.
I tried to use my BlackBerry to check the browser redirect, but it didn't work, although I made sure it is recognized as a mobile browser. What type of phone did you use to test the redirect?

Also with the station_cnt, was it meant to display the connected users or just generate sort of a log file to have some sort of statistics? The station_cnt div tag is empty and hence nothing is displayed. I have not experimented too much yet with this feature, but there is a station_cnt shell script in the PB image, couldn't that be called somehow to display the user count?

I did remove the unnecessary refresh parts from my code though, those were just leftovers from a previous chat implementation. I think I left it in the code initially, because when I tried the page without it, it didn't display correctly. Now secondary testing proved otherwise, so I cut it out. Thanks.

MaximeKjaer commented:
Hi Nargren,

I'm glad you liked it, even though it seems you have a few problems...

I used an HTC One X running Android, and an iPad (I left it out of the mobile redirects, I think that a screen that big deserves the desktop version), and everything ran smoothly. The reason for which you didn't get the redirect is probably because  your user agent string doesn't match with that one line testing for it. Come to think of it, I didn't write Symbian, PalmOS or Windows Phone in that regex... Could you try to point your BlackBerry to this jsFiddle:
http://jsfiddle.net/MaximeKjaer/HkvRt/4/
If that doesn't work, you can always try this:
http://jsfiddle.net/MaximeKjaer/HkvRt/4/embedded/result/
Or write these lines into the javascript of your piratebox:
alert(navigator.userAgent);
if (localStorage) alert("localStorage should work");

What does it return?
I'll try to test it with other devices (I have an old Symbian phone somewhere, and I know someone who has a newer Windows Phone).

As for station_cnt:
station_cnt is indeed generated by a shell script (or python, not exactly sure), but you need to load it into the page somehow, and that is where the javascript comes in. It refreshes itself so you don't have to reload the page every other minute to see how many people are connected. 
In the original PB UI, there was some javascript to load in on the page, too... It just didn't refresh automatically.
Sure, if you don't want to display it, you can just delete the function and remove the div; you'll still be able to access it by ssh'ing into your PB and opening station_cnt.txt (I think it is located at /opt/piratebox/www/station_cnt.txt).
Does it work on your BlackBerry? Otherwise I can still try to do it without jQuery, to see if that is any better...

Thank you for pointing out the mistakes!
Maxime

Nargren commented:
Hello,

I have just tried the pages with my BlackBerry, but I couldn't see any results on neither of the 2 pages. I guess it is just my old phone. I did see the code in the JavaScript so it was weird it didn't work for my phone, apparently I don't have any other phones to test it with.

Before I was trying to get something from this site: http://detectmobilebrowsers.com/ and implement it into the page, but I couldn't do it, it just didn't work. Although 3 options should be fine: python, jquery and JavaScript as well. I will try to look into it again, but I am a little short on time.

Station_cnt is just a command you can issue in the terminal to display connected users, this was then written into a shell script to make it easier to use and display. I understood that the script refreshes the file so there is no need for the JS to refresh it again, what I meant was that the div tag in the bottom of the page is actually empty:

<!-- And a div for station_cnt -->      

<div id="station_cnt"></div>

I am just about starting to get into JavaScript so I am no way an expert in that, but shouldn't the JS function be called in the station_cnt div tags? It's just because it didn't display me anything. Shouldn't the station_cnt.sh script be placed in the cgi directory to be executable? Or did you just move the station_cnt.sh script to your www folder and it worked for you?

MaximeKjaer commented:
Hi again :)

I'm beginning to think about one thing: are you sure javascript even is enabled? I'm just asking, it's usually the simplest things that are causing issues.

Station_cnt:
Are you sure you have executed these commands?

```
  /etc/init.d/cron enable && /etc/init.d/cron start 
  /opt/piratebox/bin/install_piratebox.sh /opt/piratebox/conf/piratebox.conf station_cnt 
```

That's all you should do. Otherwise it certainly won't work, and that would explain why you are experiencing problems ;) 

Otherwise, please wait 8 seconds. Maybe it only shows on the 2nd refresh?
EDIT: I tested it on my old Nokia phone. It didn't show up. I'll try to rewrite it without jQuery.
It didn't get redirected either. I guess it's just too old, but I'll try to find a solution that works on all devices.
2nd EDIT: It did work? I just waited 8 seconds. I'll edit it so that it loads station_cnt.txt on document load.

station_cnt is an optional log of the number of users. You can choose to enable it, and a server-side shell script will run on your PB every 2 minutes. It will save the number of connected users to /opt/piratebox/www/station_cnt.txt.
Then the JavaScript fetches that log: I'm pretty sure that it was intended to go on the HTML page, as it's saved in the form af an `<a>` tag.
It refreshes server-side on the PB every 2 minutes (station_cnt.txt is rewritten every other minute), but then the user needs to fetch that new data, with javascript so that he doesn't need to refresh the page to get the new number of connected clients.

And the script tag can be loaded anywhere. Some prefer to put it in the `<head>`, where it was originally intended to be (that's what I do). But recently some devs have started putting it at the bottom of the `<body>`, so that the HTML loads first, getting a UI to the users faster. The functionality is loaded as the last thing. Really, that is only a problem on very heavy webapps; where you place your `<script>` is not very relevant in this case it won't really change anything.
Theoretically, you could put it pretty much anywhere, but it is just considered a good practice to put it in the <head>

I'll try to have a look to see what other options are available to redirect mobile devices, but I guess javascript is the easiest thing to do, and probably the most efficient, too: I don't think that the browser sends it userAgent string to the server. -- In fact, javascript is probably the only option.

Maxime

Nargren commented:
Phone: Yes JavaScript is enabled, I did check to make sure before going to the jsfiddle site and the PB as well. I have just tried disabling "Terminate Slow Running Scripts", but still nothing. I think my phone is simply old...If it works on all other phones, I am happy. Plus I checked and even on my old phone the full site loaded very nicely and I could navigate easily etc. The mobile site is just an extra :D

Staation_cnt: I don't know if I am running the shell script on my router by default, I will check that tomorrow. This might be my issue.

I get the point with the placing of the scripts, I also think it is OK to leave them were they are, this page is not that heave + all the transfer is through wifi, which is much faster that an internet connection would be, plus ping is also negligible, so it will not effect the speed at which the page loads.

I will check everything again tomorrow! Thank you for the tips.

MaximeKjaer committed:
Station_cnt on document load

station_cnt now loads on document load - fixed the error where it would only load after 8 secs

MaximeKjaer committed:
sessionStorage

I believe that sessionStorage matches the idea of the PirateBox better: no data is logged, and sessionStorage is deleted at the end of the session.

MaximeKjaer committed:
sessionStorage and station_cnt loaded onload

Nargren commented:
Hey!

I have had a run again on the mobile redirect.
I couldn't make it to work with my phone, as I said its an old version with an old browser. However with some add-ons in FireFox I could set the UserAgent to Android or iPhone browser and it got redirected nicely. So it works perfectly, it just my phone the issue.

I think I will do some design tweaks on the mobile page as well, just to make it look nicer and friendlier on a handheld device as well.

I still have troubles though with the station_cnt. When I run the script it does built the txt file and tells me how many users are connected, but it isn't displayed on the index page.

MaximeKjaer committed:
Update Design2.0/index.html

typo 

MaximeKjaer commented:
Hi Nargren,

I'm really sorry that I haven't answered you until now, but I've been quite busy with exams lately :(

Your problems with station_cnt seem really weird, I don't know exactly why you don't get the same results as I do...
I'd like you to try a few things, though:
1. What's your version of the PB?
   `cat /opt/piratebox/version`
2. Could you try to check that you are running the file from here:
   `/opt/piratebox/www/`
   This command should do;
   `cat /opt/piratebox/www/index.html`
   Also, is your station_cnt located here?:
   `/opt/piratebox/www/station_cnt`
3. Maybe it's in the Javascript...
   Try to add this to index.html:

```
$.get('station_cnt.txt', function(data) {
  alert(data);
});
```

Could you also try to check your developer tools to see if there are any errors? (It's F12 on Chrome, and on Firefox I've heard about an extension called Firebug...)

Let me know about the mobile page redesign, I'd love to help if you need it !

Maxime

Nargren commented:
Hello,

it is fine, same situation here.
I'll do some additional checking over this weekend (hopefully) and try to look into station_cnt more closely. I have talked to Matthias about this as well, tried his design with station_cnt as well, but didn't work. If its just for me I don't really care to be honest.

Mobile redirect works nicely so I am happy about that. Also from Matthias again, I have tried some changes to the chat, actually the iframe can now be excluded and the chat will be a part of the index page :D. Which is just better. I would like to pull your changes first and then upgrade the chat.

Just one more thing, layout wise, you added some parts to the bottom right of the page )if I remember correctly), saying things about the box and the place for station_cnt as well. Couldn't that go into the page footer (just the bar at the bottom of the page?)? I sort of reserved some place for these things there :) the footer could basically hold anything, I just filled it up initially, so it isn't empty. Any ideas on this?

Nargren

MaximeKjaer commented:
Hi again!

Yes, taking the chat out of the iframe definitely sounds like a good
improvement, great idea!

The content in the right bar could definitely go in the footer, and would
probably be better placed there. It's true that the section I added to show
the pb's specifications is a repetition of what is in the footer, so I
guess that the div can go, too. Station_cnt would also be better placed in
the footer, so let's do that.

Would it be any better if I could create some sort of function to just get
the number of connected users? Instead of having to display "there are
currently 2 connected users" (that's the content of station_cnt.txt if I
remember correctly), we could display "2 connected users" or "2 users" ?
That might be more practical on the mobile page, at least, where the screen
space we have got to work with is smaller... It may also be insignificant,
so please tell me :)

I have another idea for what improvements could be the next goal: I've
tried to access the directory listing, but it isn't stored as an editable
HTML file (I think the HTML is stored as a string in the Python). I may
have an idea of how to find a way around that, I'll let you know if it
works. There was a good tutorial on a french blog (translated to english,
something along the lines of pboxalsace.wordpress.com) on how to achieve
this, but this was on the old version :/
This would allow us to show folder icons for directories and file-specific
icons (e.g. a Word icon for .doc or .docx files)

I've been thinking a lot about how the PirateBox as a whole could receive a
few UI improvements to make it more user- and admin-friendly. I'm not
saying that this should be done, but I just thought that I'd share my ideas
with you:
This is more of a random thought than a concrete idea: a GUI for the admin,
an options page.
This is a bit ambitious, and hasn't got much to do with the original
Design2.0 project, but it could be awesome. It would also involve a lot of
work on server-side scripts, so I really don't know if I can pull this
off... unless the options are placed in a file accessible through ssh (but
that isn't a great UI).

Another non-design-related idea is to write a cheat sheet with useful ssh
commands; I tend to forget what commands are useful, or the location of
certain files, so this could be a nice thing to have.

Let me know what you think, or if you have any other ideas!

Maxime
On Nov 28, 2012 9:49 PM, "Nargren" notifications@github.com wrote:

> Hello,
> 
> it is fine, same situation here.
> I'll do some additional checking over this weekend (hopefully) and try to
> look into station_cnt more closely. I have talked to Matthias about this as
> well, tried his design with station_cnt as well, but didn't work. If its
> just for me I don't really care to be honest.
> 
> Mobile redirect works nicely so I am happy about that. Also from Matthias
> again, I have tried some changes to the chat, actually the iframe can now
> be excluded and the chat will be a part of the index page :D. Which is just
> better. I would like to pull your changes first and then upgrade the chat.
> 
> Just one more thing, layout wise, you added some parts to the bottom right
> of the page )if I remember correctly), saying things about the box and the
> place for station_cnt as well. Couldn't that go into the page footer (just
> the bar at the bottom of the page?)? I sort of reserved some place for
> these things there :) the footer could basically hold anything, I just
> filled it up initially, so it isn't empty. Any ideas on this?
> 
> Nargren
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Nargren/PirateBox/pull/1#issuecomment-10820853.

Nargren commented:
Great ideas!
I was also thinking about directory listing/admin UI, however:
The directory listing is indeed different that in previous versions -I have also red the French guide how to change it for previous PB version, but it doesn't work with lighttpd. We have to look into the lighttpd directory listing for this. (I think there is a link somewhere on the PB forums to do something similar)

Admin thing is a viable idea, however I would prefer NOT to display it on the home screen, rather than that talk to Matthias and add an address (say 192.178.1.254) that would redirect to the admin site instead of the main PB site. SO that admins/PB owners could get to the function but not the public (or not directly at least). Also password protection here (besides router password maybe? but that's too much detail for now)
A list of commancds on the admin page could be the very first version, like how to change some settings, manage uploaded files etc.

Design2.0 project just started for designing a more developed UI for the project and it can and should evolve to become even better and have much more functionality indeed.

PS: Somehow my phone did display "There are currently 1 users connected" today when I used my box for 5 mins only:D

I red on reddit that someone tried games on PB.I was looking briefly into this and it works OK. I have tried downloading a flash game and running in through a separate page and it works like a charm. Also On my own design 2.0 (modified version- I have some test videos and audio as well, both are running fine. Question is: how many users can be supplied without problem issues? The MR3020 is quite weak in terms of processing power, so I have no idea. I think this could be a viable way to go as well, sort of audio and / or video streaming. Maybe a package for string boxes?

MaximeKjaer commented:
Well, I think that it just might be possible to get to the directory listings without even touching lighttpd :) We can load the page through an XMLHttpRequest (through the Javascript), and inject the new HTML (and hide everything else). Doing this would allow us to keep our Javascript and CSS, but load some new HTML, thus still being "in control". I'll try to have it working before the end of the weekend, it should be fairly easy.

An admin UI, however, might be harder to achieve, but totally doable and viable. I saw that the chat.html is referring to a file called psoread.py located at /cgi-bin/psoread.py . This could definitely be the right way to go: a Python script to take care of the login to the admin page, + it can probably read/write/delete files. I tried to access psoread.py to have a look, but I couldn't find it. I can't even cd to /cgi-bin!

We should probably ask Matthias if he can set up some sort of IP address or port that will take us to a login page. I was thinking that we could maybe just replicate the way  piratebox.lan/Shared is handled. I think that it is a script handling everything. How does piratebox.lan/Admin sound? :)
And yes, definitely password protection! For security reasons, I think that the password should be stored in a file which isn't accessible from the web interface. Also, let's not link to the admin panel on the front page!

But what would this admin page offer? The ability to delete and modify files would be awesome!

As for streaming content, I really don't know how to achieve that, but I think I read a forum post about that a while ago. I don't think the guy who posted it ever managed to stream content without a PC, though. But it would be a great feature to have :) The MR3020 is indeed quite weak, so I guess 3-4 users would be the maximum.

I really love the fact that browsers tend to want to open a growing number of file types on their own: on the piratebox, at least, it allows us to upload lots of different files that users can access seamlessly the files. Flash games, PDFs, audio, video... It just works, which is awesome for us!

String boxes, do you mean like this? http://www.sunways.eu/en/products/solar-inverter/accessories/can-string-box/

Nargren commented:
piratebox.lan/Admin does sound good to me, I will experiment a little with my box if I can add this IP to it to be redirected somewhere else. Shouldn't be that hard (famous last words maybe...)

Well the admin page I think should do the following:
-Display number of connected users  (station_cnt)
-Allow deleting/renaming/moving of files already uploaded
-Probably allow to change ESSID or Channel (Not sure if necessary)
-Instructions fur general functions as you said, maybe short troubleshooting pdf guide?
-Restart box feature? (again, usually not needed and most likely to be done through ssh)
-(Maybe link to somehow open a terminal window)

Streaming mediadoes work, even on the MR3020, although I have only tried it with me connected, so have no idea on limits. HTML5 is working fine so html media is good, something like:
<audio controls="controls">
<source src="music.ogg" type="audio/ogg">
<source src="music.mp3" type="audio/mpeg">
Your browser does not support the audio element, soz.
</audio>
This works with <video> as well :) I will upload my modified PB version somewhere if you want to check it out, or you can just try it yourself.

Another feature I always wanted is multiple languages. Like some websites have little flags on the top where you can select the language you want.

I will write all these things together on a list (checklist xD) so we know what we should do /what has to be done.

I think I will start everything by pulling your changes to my repository. I'm not really familiar with github, but is there no other way than pulling things back and forth?
Also, don't you think some more graphics would be nice? I was thinking about some header type things, like above the menu bar have some graphics with a PB logo and some slogan? or any other design-related idea?(this is has really nothing to do with the actual way the site works, but still. attracting people :P )

Oh and "String boxes" should have been "Strong boxes" (more CPU etc), like having a different UI for other than MR3020 boxes that can handle video or music etc.

MaximeKjaer commented:
I've been digging around in my PirateBox, too. What I did was to mount the .img file to /mnt/loop/ on my Linux computer; from there on, I could easily inspect the files. I did find the files in the cgi-bin, at chat/cgi-bin, and I've copied them over to my HDD. If it makes it easier to investigate, here are my files from pbIMG.img and pbIMG_ws.img :
http://ge.tt/9tXG1uS

It's funny that you thought about translating the files, I've had the exact same thought! I could (with a little help from some friends for some languages) translate the files into French, Danish, Estonian, and perhaps even German.
In regard of optimization, I thought that it would maybe be easier to have every translation stored in a JSON file: it would reduce the necessary storage space, and the amount of data that the PirateBox has to send. But then again, this would be a very small optimization, and it might not make a huge difference. For now, since we don't have an admin page through which we can set some settings, I think that we should have 2 files for each language:
- An HTML file ('index_fr.html'), that should be copied to opt/piratebox/www/index.html (will be set as a default)
- A JSON file ('_locales/fr/messages.json'): for whenever the user wishes to change the language. This way, the PB won't have to send the whole index.html again, and the user doesn't have to reload the page.

This is how it is done in Google Chrome extensions:
https://developer.chrome.com/extensions/i18n.html

I'm also pretty new to Github, but this tutorial helped me out on the subject,
https://help.github.com/articles/using-pull-requests

I really like the idea of not having to limit ourselves and make a package for stronger boxes. However, all optimizations that are made with the MR3020 in mind and which don't affect the user experience negatively should stay in a stronger box.

Graphics would be nice indeed! Sadly, my designing skills are well below par, so I can't really make anything beautiful... I'll try to think about it though.

I have succeeded in running a Python script through my browser's call, and the script will eventually run a file that I have called reboot.sh. I need to get some further testing donce, but this should be it. I don't know where to store the password, though... I saw that I could also get the uptime, the version (there's a command called `/sbin/uptime` and another is something along the lines of `cat /etc/piratebox/version`) I know where to find a whole lot of good info, at least!

Nargren commented:
I have talked to Matthias about the admin page, it could be something like admin.piratebox.lan (maybe) http://redmine.lighttpd.net/projects/lighttpd/wiki/Docs_ModSimpleVhost but I seriously have to dig into this. Matthias said if i can make it work, it would be no problem to implement it as default for later PirateBox versions.

Translating into languages shouldn't be that difficult, the ones you said could be easily managed with the joint help of the forum people for example. I can also give some help with french, german and hungarian. The more languages we can add, the more attractive it will be to everybody .
However, I am not familiar whatsoever with this type of website editing, so need to read into it.

I am far from being a designer as well, but I will ask around more talented friends for design elements.

I gave a little thought on how to rename or edit files from within the browser and i don't know how easy that would be. Scripts can be run from the browser, but then inputs have to be given to those scripts (say a file, saved from the admin page?)I will logically thing through again this thing, but it seems tricky.

I really don't have much time (and will not have much in the following weeks either), but I will try to dedicate some to
1. Make or Ask someone for a matching design element(s) for the UI to make it more attractive
2. Think through an admin page like admin.piratebox.lan and try to make it work on a PB
3. Make a concept for editing files from within this admin page & test
4. Shall I have excessive free time, I will start translating the text into foreign languages and save them as a text file for further steps of the project.

MaximeKjaer commented:
Hi Nargren!

I've been trying find out how to implement the SimpleVhost, but I couldn't really find what file to modify. So far, my best guess is `/conf/lighttpd/lighttpd.conf`, but it only exists in my .img file, not on my actual PirateBox.

I think I may have managed to find how the admin should log in, though.

First of all, we can have a file called login.html. It talks with login.py, which is located at `/opt/piratebox/www/cgi-bin/login.py`. The username and password are hard coded into login.py: if the username and password sent by login.html are wrong, the Python file will respond "false", and the HTML page will alert the user. If the login information is correct, login.py  copies `admin_tools.py.bak` to `admin_tools.py`; the user stores his password in sessionStorage, and the Javascript redirects him to admin.html.

When admin.html is opened, it sends the username and password stored in sessionStorage.
`admin_tools.py` can now offer him a series of commands, which are executed by shell scripts, called by the Python.

On logout, sessionStorage and admin_tools.py are deleted (we still have admin_tools.py.bak).

Right now, I'll just run the code from piratebox.lan/admin.html, but if we could figure out how to get a domain name to redirect to this, it would just be plain awesome.

Deleting files shouldn't be too hard, if we can get the html to send data that the python will understand.
http://stackoverflow.com/questions/1995373/deleting-files-in-python

Also, have a look at these files:
http://ge.tt/1F4kv3T/v/0
This is what was in the cgi-bin.

So far, I have managed to create a small Python script which returns uptime and cpuinfo (bot located at /proc/) to the user. No logins and no security yet, but that should probably be the last step.

Maxime

Nargren commented:
Hi,
sorry for not getting back to you for such a long time, just been through exams.
I'm still a little busy, but just wanted to let you know that I finally sat down and pulled your changes.
Also, I have improved the chat so that its no longer an iframe, but part of the main page (from Matthias) for both browser and mobile version!

Nice one with the admin login page!

Deleting files wouldn't be hard, IF we know which file we are talking about, since we can send commands through website. Like you can write a code into the html page that will delete file1 if you click a link or something. But that could only delete file1 and not file2, so there should be some sort of input (I'm thinking about a textbox or dropdown menu listing maybe) which seems to me a little bit more difficult for me as for the moment, but sure it can be done.

"Right now, I'll just run the code from piratebox.lan/admin.html, but if we could figure out how to get a domain name to redirect to this, it would just be plain awesome." 
I will be looking into this as soon as I will have some spare time as well.

MaximeKjaer commented:
Hi again!

No problem, I know what it's like... I have exams coming up in January, too :/

I have tried to develop a solution for the admin tools, and I have uploaded it on my fork. So far, I'm able to send data to the HTML (admin.py sends to admin.html).
I'm sending everything in a JSON file to reduce the load on the box. As you can see, there's a lot more Javascript than Python, and that is done one purpose; the more I can get the client to do, the better (also, I'm better at JS). Currently, the Python sends information about the box, uptime, and info about the wireless settings. I've gotten the Javascript to sort through the wireless file and replace the SSID with what is entered in a textbox. I can't get the python to read what is coming back, but I'm still trying to copy the behaviour of psowrte.py.

For file deletion, I thought that we could just use an iframe with the file directory, and add a cross for each line through JS. The JS would then know what file to delete, and could send the data to a python file (or even .sh if it's possible to pass variables).

I'll have some time to work on it as I'm finally on vacation (although I have loads of work to do, exams to practice for...)
